File: Insertion_Sort/Ruby/mitogh/insertion_sort.rb
def insertion_sort(arr = [])    
  arr.each_with_index do |element, i| 
    j = i
    while(j > 0 and (prev = arr[j-1]) > element) do
      arr[j] = prev
      j -= 1
    end
    arr[j] = element
  end
end

File: Insertion_Sort/PHP/mitogh/insertion_sort.php
<?php
function insertion_sort($arr = []){
    $n = count($arr);

    for($i = 0; $i < $n; $i++){
        $element = $arr[$i];
        $j = $i; 
        while($j > 0 && $arr[$j-1] > $element){
            $arr[$j] = $arr[$j-1];
            $j--;
        }
        $arr[$j] = $element;
    }
    return $arr;
}
